root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1609,6 +1609,13 @@ const testData = [
             expected_clean_url: 'https://itunes.apple.com/artist/id444923726',
        only_valid_entity_types: ['artist'],
   },
+  {
+                     input_url: 'https://itunes.apple.com/us/author/paige-lewis/id348965238',
```

Comment:
Adding a fragment would allow to detect the above mentioned issue:
```suggestion
                     input_url: 'https://itunes.apple.com/us/author/paige-lewis/id348965238#',
```